app/models/deploy.rb

```diff
@@ -110,7 +110,7 @@ def self.finished_naturally
   end
 
   def self.prior_to(deploy)
-    deploy.persisted? ? where("#{table_name}.id < ?", deploy.id) : self
+    deploy.persisted? ? where("#{table_name}.id < ?", deploy.id) : all
```

Comment:
`.where(nil)` seems to be best / does not add an extra condition

On Wed, Mar 11, 2015 at 3:41 PM, Jon Moter <notifications@github.com> wrote:

> In app/models/deploy.rb
> <https://github.com/zendesk/samson/pull/289#discussion_r26263617>:
>
> > @@ -110,7 +110,7 @@ def self.finished_naturally
> >    end
> >
> >    def self.prior_to(deploy)
> > -    deploy.persisted? ? where("#{table_name}.id < ?", deploy.id) : self
> > +    deploy.persisted? ? where("#{table_name}.id < ?", deploy.id) : all
>
> To be fair, calling #all turns it into an array instead of an arel, which
> would interfere with chaining. So it's probably better to use where('1=1'),
> unless you know of another method that returns an arel but preserves
> chaining.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/samson/pull/289/files#r26263617>.
>